frontend/src/app/settings/core/user-settings/user-settings.component.ts

```diff
@@ -114,15 +115,17 @@ export class UserSettingsComponent implements OnInit {
 
     dialogRef.afterClosed().subscribe((result: InputDialogResult) => {
       if (result.success && result.text) {
-        this.userService.createUser(username, 'user', result.text).subscribe({
-          next: () => {
-            this.toastService.showSuccess(
-              'User created',
-              `The user ${username} has been created.`,
-            );
-            this.getUsers();
-          },
-        });
+        this.usersService
+          .createUser({ name: username, role: Role.User, reason: result.text })
+          .subscribe({
+            next: () => {
+              this.toastService.showSuccess(
+                'User created',
+                `The user ${username} has been created.`,
+              );
+              this.getUsers();
```

Comment:
It might be worth adding a user wrapper service so that we don't have to make all these `this.getUsers()` calls implicitly, but have some behavior that keeps users up to date. However, we should check if this additional abstraction is necessary or if we can either keep the current approach or find some other approach (i.e. I haven't looked into signals yet, but they might be useful here).

Reply:
Makes sense, but is out of scope for this PR.